File: backend/app/models/dispositivo.py
from sqlalchemy import ForeignKey, Integer, String
from sqlalchemy.orm import relationship

from app.extensions import db


class Dispositivo(db.Model):
    __tablename__ = "dispositivos"

    id_dispositivo = db.Column(Integer, primary_key=True)
    codigo_dispositivo = db.Column(String(50), unique=True, nullable=False)
    modelo = db.Column(String(100))
    estado = db.Column(String(20), nullable=False)
    id_puesto = db.Column(Integer, ForeignKey("puestos.id_puesto"), nullable=False)

    puesto = relationship("Puesto", back_populates="dispositivos", lazy="select")
    asistencias = relationship("Asistencia", back_populates="dispositivo", lazy="select")

    def __repr__(self) -> str:
        return f"<Dispositivo {self.codigo_dispositivo}>"

File: mobile/lib/features/devices/models/dispositivo.dart
class Dispositivo {
  const Dispositivo({
    this.idDispositivo,
    required this.codigoDispositivo,
    required this.modelo,
    required this.estado,
    required this.idPuesto,
    this.puestoNombre,
  });

  final int? idDispositivo;
  final String codigoDispositivo;
  final String? modelo;
  final String estado;
  final int idPuesto;
  final String? puestoNombre;

  factory Dispositivo.fromJson(Map<String, dynamic> json) {
    return Dispositivo(
      idDispositivo: json['id_dispositivo'] as int?,
      codigoDispositivo: json['codigo_dispositivo'] as String,
      modelo: json['modelo'] as String?,
      estado: json['estado'] as String,
      idPuesto: json['id_puesto'] as int,
      puestoNombre:
          (json['puesto'] as Map<String, dynamic>?)?['nombre_puesto']
              as String?,
    );
  }

  Map<String, dynamic> toJson() => {
    'codigo_dispositivo': codigoDispositivo,
    'modelo': modelo,
    'estado': estado,
    'id_puesto': idPuesto,
  };
}
